test/signs_ui_web/controllers/messages_controller_test.exs

```diff
@@ -0,0 +1,35 @@
+defmodule SignsUiWeb.MessagesControllerTest do
+  use SignsUiWeb.ConnCase
+
+  @update_attrs %{
+      "MsgType" => "SignContent",
+      "c" => "e120~m2-\"Bowdoin 4 min\"",
+      "sta" => "BGOV",
+      "uid" => 22
+    }
+
+  describe "index" do
+    test "lists all messages", %{conn: conn} do
+      conn = conn
+             |> add_req_header(:basic)
+             |> get(messages_path(conn, :index))
+      assert html_response(conn, 200) =~ "Listing Messages"
+    end
+  end
+
+  describe "create messages" do
+    test "responds with the json message when its valid", %{conn: conn} do
```

Comment:
Can you also have one where the API key isn't valid?

Reply:
i can